tests/core/test_featurizer.py

```diff
@@ -1,111 +1,284 @@
+    encoded_actions = f.encode_all_actions(domain, None)
+    assert len(encoded_actions) == len(domain.action_names)
+    assert all(
+        [
+            ACTION_NAME in encoded_action and ACTION_TEXT not in encoded_action
+            for encoded_action in encoded_actions
+        ]
+    )
 
 
-def test_label_tokenizer_featurizer_handles_on_non_existing_features():
-    f = LabelTokenizerSingleStateFeaturizer()
-    f.user_labels = ["a_d"]
-    f.bot_labels = ["c_b"]
-    f.user_vocab = {"a": 0, "d": 1}
-    f.bot_vocab = {"b": 1, "c": 0}
-    f.num_features = len(f.user_vocab) + len(f.slot_labels) + len(f.bot_vocab)
-    encoded = f.encode(
-        {"a_d": 1.0, "prev_c_b": 0.0, "e": 1.0, "prev_action_listen": 1.0}
+def test_single_state_featurizer_uses_dtype_float():
+    f = SingleStateFeaturizer()
+    f._default_feature_states[INTENT] = {"a": 0, "b": 1}
+    f._default_feature_states[ACTION_NAME] = {"e": 0, "d": 1}
+    f._default_feature_states[ENTITIES] = {"c": 0}
+    encoded = f.encode_state(
+        {
+            "user": {"intent": "a", "entities": ["c"]},
+            "prev_action": {"action_name": "d"},
+        },
+        interpreter=None,
     )
-    assert (encoded == np.array([1, 1, 0, 0])).all()
-
-
-def test_label_tokenizer_featurizer_uses_correct_dtype_int():
-    f = LabelTokenizerSingleStateFeaturizer()
-    f.user_labels = ["a_d"]
-    f.bot_labels = ["c_b"]
-    f.user_vocab = {"a": 0, "d": 1}
-    f.bot_vocab = {"b": 1, "c": 0}
-    f.num_features = len(f.user_vocab) + len(f.slot_labels) + len(f.bot_vocab)
-    encoded = f.encode({"a_d": 1.0, "prev_c_b": 0.0, "prev_action_listen": 1.0})
-    assert encoded.dtype == np.int32
-
-
-def test_label_tokenizer_featurizer_uses_correct_dtype_float():
-    f = LabelTokenizerSingleStateFeaturizer()
-    f.user_labels = ["a_d"]
-    f.bot_labels = ["c_b"]
-    f.user_vocab = {"a": 0, "d": 1}
-    f.bot_vocab = {"b": 1, "c": 0}
-    f.num_features = len(f.user_vocab) + len(f.slot_labels) + len(f.bot_vocab)
-    encoded = f.encode({"a_d": 0.2, "prev_c_b": 0.0, "prev_action_listen": 1.0})
-    assert encoded.dtype == np.float64
-
-
-def test_label_tokenizer_featurizer_handles_on_non_existing_probabilistic():
-    f = LabelTokenizerSingleStateFeaturizer()
-    f.user_labels = ["a_d"]
-    f.bot_labels = ["c_b"]
-    f.user_vocab = {"a": 0, "d": 1}
-    f.bot_vocab = {"b": 1, "c": 0}
-    f.num_features = len(f.user_vocab) + len(f.slot_labels) + len(f.bot_vocab)
-    encoded = f.encode(
-        {"a_d": 0.2, "prev_c_b": 1.0, "c": 0.0, "e": 1.0, "prev_action_listen": 1.0}
+    assert encoded[ACTION_NAME][0].features.dtype == np.float32
+
+
+def test_single_state_featurizer_correctly_encodes_state_with_interpreter(
+    monkeypatch: MonkeyPatch, tmp_path: Path, unpacked_trained_moodbot_path: Text
+):
+    # Skip actual NLU training and return trained interpreter path from fixture
+    _train_nlu_with_validated_data = Mock(return_value=unpacked_trained_moodbot_path)
```

Comment:
why do we need to mock the training here? Can't we just load the interpreter from the trained model? Instead of retraining to load the existing trained interpreter? 😆